appjs/templates/theme_list.ejs

```diff
@@ -53,59 +59,135 @@
+          <li class=""><a href="<%=getPageUrl(collection.state.firstPage)%>"><%=collection.state.firstPage %><span class="sr-only"></span></a></li>
+        <% } %>
+        <% if(collection.state.currentPage - 3 > collection.state.firstPage){ %>
+          <li class="ellipses"><span>...</span></li>
+        <% } %>
+
+        <% if(collection.state.currentPage === collection.state.firstPage){ %>
+
+          <% var i = 1; %>
+          <li class="active"><a href="#"><%=collection.state.currentPage %><span class="sr-only">(current)</span></a></li>
+          <% while(collection.state.firstPage + i <= collection.state.totalPages && i < 6) { %>
+            <li class=""><a href="<%=getPageUrl(collection.state.currentPage + i)%>"><%=collection.state.currentPage + i %><span class="sr-only"></span></a></li>
+            <% i++; %>
+          <% } %>
+
+        <% } else if(collection.state.currentPage === collection.state.lastPage){ %>
+
+          <% var i = -5; %>
+          <% while(i < 0) { %>
+            <% if(collection.state.lastPage + i > collection.state.firstPage){ %>
+              <li class=""><a href="<%=getPageUrl(collection.state.lastPage + i)%>"><%=collection.state.lastPage + i %><span class="sr-only"></span></a></li>
+            <% } %>
+            <% i++; %>
+          <% } %>
+          <li class="active"><a href="#"><%=collection.state.currentPage %><span class="sr-only">(current)</span></a></li>
+
+        <% } else { %>
+
+          <% var i = -2; %>
+          <% while(i < 0) { %>
+            <% if(collection.state.currentPage + i > collection.state.firstPage){ %>
+            <li class=""><a href="<%=getPageUrl(collection.state.currentPage + i)%>"><%=collection.state.currentPage + i %><span class="sr-only"></span></a></li>
+            <% } %>
+            <% i++; %>
+          <% } %>
+
+          <li class="active"><a href="#"><%=collection.state.currentPage %><span class="sr-only">(current)</span></a></li>
+
+          <% var i = 1; %>
+          <% while(collection.state.currentPage + i <= collection.state.lastPage && i < 3) { %>
+            <li class=""><a href="<%=getPageUrl(collection.state.currentPage + i)%>"><%=collection.state.currentPage + i %><span class="sr-only"></span></a></li>
+            <% i++; %>
+          <% } %>
+
+        <% } %>
+
+
+        <% if(collection.state.lastPage - collection.state.currentPage > 2){ %>
+          <% if(collection.state.lastPage - collection.state.currentPage > 3){ %>
+          <li class="ellipses"><span>...</span></li>
+          <% } %>
+          <li class=""><a href="<%=getPageUrl(collection.state.lastPage)%>"><%=collection.state.lastPage %><span class="sr-only"></span></a></li>
+        <% } %>
+
+      </ul>
+
+      <% if(collection.state.currentPage !== collection.state.lastPage){ %>
+        <li class="next<%=hasNextPage() ? '' : ' disabled' %>"><a href="<%=getNextPageUrl()%>">Next</a></li>
+      <% } %>
+    </ul>
+  </nav>
+  <% } %>
```

Comment:
Oh - I take that back, kind of. I got confused as to which comment I was replying to... apparently replying on my phone while on a train on a bridge didn't work out so well. 

I modified the classes that were already there for the most part, but did have add to them a bit; plus, I had to create the extra buttons since bootstrap doesn't do that.